### PR TITLE
adds e2e CI test for hypertrace-ingester

### PR DESCRIPTION
**Pull request by JBAhire**
This PR,
- will add quick e2e test for hypertrace-ingester which will 
   - Build image on each new PR and push to main branch
   - test if we are able to get full stack up and running
   - generate and retrieve trace.

### CODE REVIEW

File: .github/workflows/hypertrace-ingester/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      kafka-zookeeper:
+        condition: service_healthy
+
+  hypertrace-ingester:
+    image: hypertrace/hypertrace-ingester:test
+    container_name: hypertrace-ingester
+    environment:
+      - KAFKA_BOOTSTRAP_SERVERS=kafka:9092
+      - DEFAULT_TENANT_ID=__default
+      - SPAN_GROUPBY_SESSION_WINDOW_INTERVAL=2
```
Comment (adriancole) [inline .github/workflows/hypertrace-ingester/docker-compose.yml:46]: these defaults really need to go upstream as they are copy/pasta

File: .github/workflows/hypertrace-ingester/scripts/tests.sh [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  fi
+
+  RETRY_COUNT=$((RETRY_COUNT+1)) 
+  sleep 3
+done
+
+echo ""
+echo "Found trace \"$TRACE_ID\"."
+
+echo "Checking for broken links"
```
Comment (adriancole) [inline .github/workflows/hypertrace-ingester/scripts/tests.sh:96]: once this is in we should remove it from hypertrace/hypertrace

File: .github/workflows/docker-tests.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
+name: e2e test
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
```
Comment (adriancole) [inline .github/workflows/docker-tests.yml:7]: comment what this dispatches to?

**Review by adriancole — APPROVED** [review]
thanks for doing this

File: .github/workflows/hypertrace-ingester/docker-compose.yml (hunk 1 above)
Comment (JBAhire) [reply .github/workflows/hypertrace-ingester/docker-compose.yml:46]: @kotharironak , how are we planning to move this upstream?

File: .github/workflows/hypertrace-ingester/scripts/tests.sh (hunk 2 above)
Comment (JBAhire) [reply .github/workflows/hypertrace-ingester/scripts/tests.sh:96]: cool

File: .github/workflows/docker-tests.yml (hunk 3 above)
Comment (JBAhire) [reply .github/workflows/docker-tests.yml:7]: okay!

File: .github/workflows/docker-tests.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+name: e2e test
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  # workflow_dispatch will let us manually trigger the workflow from GitHub actions dashboard.
```
Comment (adriancole) [inline .github/workflows/docker-tests.yml:7]:
```suggestion
  # workflow_dispatch will let us manually trigger the workflow from GitHub actions dashboard.
  # See https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/manually-running-a-workflow
```

File: .github/workflows/docker-tests.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        uses: actions/cache@v1
+        with:
+          path: ~/.gradle
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/build.gradle.kts') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Invoke Docker build with Gradle
+        uses: eskatos/gradle-command-action@v1
+        with:
+          arguments: dockerBuildImages
+      - name: verify docker image creation
```
Comment (adriancole) [inline .github/workflows/docker-tests.yml:50]: remove this as no longer debugging the job

File: .github/workflows/hypertrace-ingester/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        aliases:
+          - jaeger
+          - jaeger-collector
+          - zipkin
+    depends_on:
+      kafka-zookeeper:
+        condition: service_healthy
+
+  hypertrace-ingester:
+    image: hypertrace/hypertrace-ingester:test
```
Comment (adriancole) [inline .github/workflows/hypertrace-ingester/docker-compose.yml:41]: sut should be this container. We shouldn't forward reference containers that depend on us

File: .github/workflows/hypertrace-ingester/scripts/tests.sh (hunk 2 above)
Comment (jcchavezs) [reply .github/workflows/hypertrace-ingester/scripts/tests.sh:96]: I think we might not need it here. It makes more sense in the `hypertrace-service` so I'd rather remove it from the ingester.

File: .github/workflows/hypertrace-ingester/docker-compose.yml (hunk 6 above)
Comment (JBAhire) [reply .github/workflows/hypertrace-ingester/docker-compose.yml:41]: done

File: .github/workflows/hypertrace-ingester/scripts/tests.sh (hunk 2 above)
Comment (JBAhire) [reply .github/workflows/hypertrace-ingester/scripts/tests.sh:96]: cool

**Review by adriancole — APPROVED** [review]

File: .github/workflows/hypertrace-ingester/docker-compose.yml (hunk 1 above)
Comment (jcchavezs) [reply .github/workflows/hypertrace-ingester/docker-compose.yml:46]: I think
- SPAN_GROUPBY_SESSION_WINDOW_INTERVAL
- REPLICATION_FACTOR
- NUM_STREAM_THREADS
- PRE_CREATE_TOPICS
- PRODUCER_VALUE_SERDE=org.hypertrace.core.kafkastreams.framework.serdes.GenericAvroSerde

can go upstream, the others sound very adhoc.

**Pull request merged**